sdk/core/azure-core/azure/core/tracing/abstract_span.py

```diff
@@ -43,28 +43,29 @@ def finish(self):
         pass
 
     @abstractmethod
-    def to_header(self, headers):
+    def to_header(self):
         # type: (Dict[str, str]) -> Dict[str, str]
         """
         Returns a dictionary with the header labels and values.
         """
         pass
 
+    @property
     @abstractmethod
-    def from_header(self, headers):
-        # type: (Dict[str, str]) -> Any
+    def span_instance(self):
+        # type: () -> Any
         """
-        Given a dictionary returns a new tracer with the span context
-        extracted from that dictionary.
+        Returns the span the class is wrapping.
         """
         pass
 
-    @property
+    @classmethod
     @abstractmethod
-    def span_instance(self):
-        # type: () -> Any
+    def from_header(cls, headers):
+        # type: (Dict[str, str]) -> Any
```

Comment:
Could the return type be more specific?

Reply:
Not at the moment. This would be used for eventhubs if at all and all of eventhubs is under questions at the moment. For now am assuming will return a tracer. Could possible also return a dictionary in the future of the span context which would have a different type depending on the implementation.

Reply:
Another point of confusion here is that the monitoring folks want me to `link` the incoming trace with the span I am already in and there are implementations that do not support that. No matter what, I know `from_headers` is going to be required, but what it returns is a question because of that.

Reply:
us now link